mmdet3d/ops/roipoint_pool3d/roipoint_pool3d.py

```diff
@@ -0,0 +1,93 @@
+from torch import nn as nn
+from torch.autograd import Function
+
+from . import roipoint_pool3d_ext
+
+
+class RoIPointPool3d(nn.Module):
+
+    def __init__(self, num_sampled_points=512, pool_extra_width=1.0):
+        super().__init__()
+        self.num_sampled_points = num_sampled_points
+        self.pool_extra_width = pool_extra_width
+
+    def forward(self, points, point_features, boxes3d):
+        """
+        Args:
+            points: (B, N, 3)
+            point_features: (B, N, C)
+            boxes3d: (B, M, 7), [x, y, z, dx, dy, dz, heading]
+
+        Returns:
+            pooled_features: (B, M, 512, 3 + C)
+            pooled_empty_flag: (B, M)
+        """
+        return RoIPointPool3dFunction.apply(points, point_features, boxes3d,
+                                            self.pool_extra_width,
+                                            self.num_sampled_points)
+
+
+def enlarge_box3d(boxes3d, extra_width=(0, 0, 0)):
```

Comment:
We have OP to enlarge the box https://github.com/open-mmlab/mmdetection3d/blob/907bb4e4d74de9e51cc1a65d88f15e08722d149b/mmdet3d/core/bbox/structures/lidar_box3d.py#L242. It should not be done in this OP

Reply:
However this op is based on lidar_box3d, the enlarge_box3d's input is a tensor, so I add this ops in `roipoint_pool3d.py`